lab_1_keywords_tfidf/main.py

```diff
@@ -3,108 +3,164 @@
 def clean_and_tokenize(text: str) -> Optional[list[str]]:
     """
-    Removes punctuation, casts to lowercase, splits into tokens
-
-    Parameters:
-    text (str): Original text
-
-    Returns:
-    list[str]: A sequence of lowercase tokens with no punctuation
-
-    In case of corrupt input arguments, None is returned
-    """
-    pass
+       Removes punctuation, casts to lowercase, splits into tokens
+       Parameters:
+       text (str): Original text
+       Returns:
+       list[str]: A sequence of lowercase tokens with no punctuation
+       In case of corrupt input arguments, None is returned
+       """
+    if isinstance(text, str):
+        target_text = text.lower().strip()
+        symbols = """!()-[]{};?@#$%:'",.^&;*_><"""
+        for symbol in target_text:
+            if symbol in symbols:
+                target_text = target_text.replace(symbol, '')
+        tokens = target_text.split()
+        for word in tokens:
+            if word == "' '":
+                tokens.remove(word)
+        return tokens
+    return None
 
 
 def remove_stop_words(tokens: list[str], stop_words: list[str]) -> Optional[list[str]]:
     """
-    Excludes stop words from the token sequence
-
-    Parameters:
-    tokens (List[str]): Original token sequence
-    stop_words (List[str]: Tokens to exclude
-
-    Returns:
-    List[str]: Token sequence that does not include stop words
-
-    In case of corrupt input arguments, None is returned
-    """
-    pass
+       Excludes stop words from the token sequence
+       Parameters:
+       tokens (List[str]): Original token sequence
+       stop_words (List[str]: Tokens to exclude
+       Returns:
+       List[str]: Token sequence that does not include stop words
+       In case of corrupt input arguments, None is returned
+       """
+    if isinstance(tokens, list) and isinstance(stop_words, list):
+        if (isinstance(word1, str) for word1 in stop_words) and (isinstance(word2, str) for word2 in tokens):
+            new_tokens = []
+            for word3 in tokens:
+                if word3 not in stop_words:
+                    new_tokens += [word3]
+            tokens = new_tokens
+            return tokens
+        return None
+    return None
 
 
 def calculate_frequencies(tokens: list[str]) -> Optional[dict[str, int]]:
     """
-    Composes a frequency dictionary from the token sequence
-
-    Parameters:
-    tokens (List[str]): Token sequence to count frequencies for
-
-    Returns:
-    Dict: {token: number of occurrences in the token sequence} dictionary
-
-    In case of corrupt input arguments, None is returned
-    """
-    pass
+       Composes a frequency dictionary from the token sequence
+       Parameters:
+       tokens (List[str]): Token sequence to count frequencies for
+       Returns:
+       Dict: {token: number of occurrences in the token sequence} dictionary
+       In case of corrupt input arguments, None is returned
+       """
+    if isinstance(tokens, list):
```

Comment:
revert the condition and return None immediately

Reply:
done